### PR TITLE
Keck/NIRES dither pattern

### PR DESCRIPTION
**Pull request by debora-pe**
This addresses the PN-15 and PN-16 requirements in JIRA for NIRES.

Added also a new dataset and unit test in the Dev Suite: https://github.com/pypeit/PypeIt-development-suite/pull/228

### CODE REVIEW

**Comment by debora-pe** [comment]
Tests pass:

```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  201 passed, 1271 warnings in 166.89s (0:02:46) ---
--- PYTEST UNIT TESTS PASSED  119 passed, 1887 warnings in 1952.87s (0:32:32) ---
--- PYTEST VET TESTS PASSED  25 passed, 1501 warnings in 816.86s (0:13:36) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 154/154 TESTS  ---
Testing Started at 2022-11-12T11:22:46.569630
Testing Completed at 2022-11-13T00:07:07.385878
Total Time: 12:44:20.816248
```

**Review by jhennawi — APPROVED** [review]
Thanks for this @debora-pe. The fixes to the frame typing and docs were badly needed.

File: doc/dev/niresconfig.rst [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+--------------------------
+
+PypeIt uses the concept of a "calibration group" to define a complete set of
+calibration frames (e.g., arcs, flats) and the science frames to which these calibration
+frames should be applied.
+
+By default, :ref:`pypeit_setup` uses the setup identifier (e.g., A,B,C,D...) to assign frames to a single calibration
+group. Since NIRES has only one configuration, i.e., ony one setup identifier, all the frames would have
+the same PypeIt keyword ``calib``. However, since it is likely, during an observing night, to observe different
+targets, PypeIt automatically assigns different ``calib`` values to the science/standard NIRES frames of different targets.
```
Comment (kbwestfall) [inline doc/dev/niresconfig.rst:50]: Is this because the science/standard frames are also used as arc/tilt frames?

File: doc/dev/niresframes.rst [hunk 2]
```diff
@@ -36,27 +37,30 @@ required for the frame-typing and their associated keyword in the
...
+``pixelflat``      ``'domeflat'``                                               Not used
+``trace``          ``'domeflat'``                                               Not used
+``arc``            ``'Object'``, ``'object'``                                   Not used
+``tilt``           ``'Object'``, ``'object'``                                   Not used
+================   ==========================================================   =========
 
 .. note::
 
-    - Frames with type ``bias``, ``illumflat``, and ``pinhole`` are *not* typed.
+    - Frames with type ``bias``, ``dark``, ``illumflat``, and ``pinhole`` are *not* typed.
```
Comment (kbwestfall) [inline doc/dev/niresframes.rst:59]: Not for this PR, but do people typically take darks with NIRES data?  If so, is there a way to automatically identify them?

**Review by kbwestfall — APPROVED** [review]
Nice!  I just had a couple of comments on the docs.

File: doc/dev/niresframes.rst (hunk 2 above)
Comment (profxj) [reply doc/dev/niresframes.rst:59]: @jhennawi -- what's the answer?

File: doc/dev/niresconfig.rst (hunk 1 above)
Comment (kbwestfall) [reply doc/dev/niresconfig.rst:50]: Added a TODO comment in the file.

File: doc/dev/niresframes.rst (hunk 2 above)
Comment (kbwestfall) [reply doc/dev/niresframes.rst:59]: Issue #1509 added.

**Pull request merged**